### ISSUE TITLE
nfsd: Unexpected status SERVICE_PAUSED in response to START control.

### ISSUE THREAD

**Issue by mac-dev** [drud/ddev#2593]
If I start 
```
$ sudo nssm restart nfsd
nfsd: Unexpected status SERVICE_PAUSED in response to START control.
```

I have edited the .ddev/nfs-exports.txt. It looks like that:

```
# Exports for winnfsd for ddev
# You can edit these yourself to match your workflow
# But nfs must share your project directory
# Additional lines can be added for additional directories or drives.
D:\ > /seu/source
```
when I run a ddev debug nfsmount It returns:
`NFS does not seem to be set up yet, see debugging instructions at https://ddev.readthedocs.io/en/stable/users/performance/#debugging-ddev-start-failures-with-nfs_mount_enabled-true
Details: error=failed to StartContainer: API error (500): error while mounting volume '/var/lib/docker/volumes/testnfsmount/_data': failed to mount local volume: mount :/d/seu/source/hessenweb:/var/lib/docker/volumes/testnfsmount/_d
ata, data: addr=192.168.65.2,hard,nolock: connection refused
output=
`

but when I start winnfsd "d:\\" it seems to work and ddev debug nfsmount returns:
```
TARGET    SOURCE                   FSTYPE OPTIONS
/nfsmount :/d/seu/source/hessenweb nfs    rw,relatime,vers=3,rsize=32768,wsize=32768,namlen=255,hard,nolock,proto=tcp,timeo=600,retrans=2,sec=sys,mountaddr=192.168.65.2,mountvers=3,mountproto=tcp,local_lock=all,addr=192.168.65.2
/nfsmount/.ddev

Successfully accessed NFS mount of D:\seu\source\hessenweb
nfs_mount_enabled is not set either globally or in this project.
Use `ddev config global --nfs-mount-enabled` to enable it.
```

What looks okay so far.

What am I doing wrong? Is there an Issue with the nfs-exports.txt? 
Somehow we do have sort of 2 Home Dirs, one which is a Networkmount (I:/.ddev) and one locally (c:/users/name/.ddev) and the nfs-exports.txt are not the same files (but same contents inbetween).

Any help greatly apreciated

**Comment by rfay** [comment]
Hi there - It sounds like you've been doing the right strategies.

I note that when you run `winnfsd "d:\"` it works; you're using lower-case there and sharing the entire D:\ drive.

You also mention that the file you've provided is .ddev/nfs-exports.txt... But it should be ~/.ddev/nfs_exports.txt (with an underscore as opposed to a hyphen... is that the problem here?)

A normal ~/.ddev/nfs_exports.txt created by the script might look like this:

```
C:\Users\rfay > /c/Users/rfay   
```

I recommend that you move back to something like that. 

At least start with the simplest thing you can do. Start with the C: drive, with something like I have, and see if you can get it to work.

Then add the D: drive, but do it with the same pathing, like 

D:\ > /d

You can do this any way you want to, and make NFS work many different ways, but you have to take it one step at a time and make sure you understand what you're doing. 

If I were you I'd delete all related files in ~/.ddev, use `sudo nssm remove nfsd`, then run the windows nfs setup script again. Get that to work. Then add additional lines to accomplish what you want. You may want to look at the (rather minimalistic) instructions at https://github.com/winnfsd/winnfsd to see how you can do this. When you're able to run winnfsd manually with your config file you should be about there.

**Comment by rfay** [comment]
No response from original poster, so closing. Happy to continue the conversation here or elsewhere.

**Comment by mac-dev** [comment]
I actually did some tests yesterday and I guess there are some kind of networking restriction that interfere here. 
Creation of a service with -addr leeds to the result described above:

```
C:\>nssm install nfsd winnfsd -id 1000 1000 -addr 192.168.65.2 "D:\seu\source" /source
Service "nfsd" installed successfully!

C:\>nssm restart nfsd
nfsd: STOP: 
nfsd: Unexpected status SERVICE_PAUSED in response to START control.
```

Without addr:

```
C:\>nssm install xyz winnfsd -id 1000 1000 "D:\seu\source" /source2
Service "xyz" installed successfully!

C:\>nssm restart xyz
xyz: STOP: the service has not been started
xyz: START: service succesfully started.
```
I've tested it without running cisco vpn and even completely offline with the same results.
If I find out what interferes here I'll drop a note

**Comment by mac-dev** [comment]
It looks like portmap service is missing. Running 

```
winnfsd -addr 192.168.65.2 "D:\" /source3 -vvv
=====================================================
WinNFSd 2.4.0 [5f7f224]
Network File System server for Windows
Copyright (C) 2005 Ming-Yang Kao
Edited in 2011 by ZeWaren
Edited in 2013 by Alexander Schneider (Jankowfsky AG)
Edited in 2014 2015 by Yann Schepens
Edited in 2016 by Peter Philipp (Cando Image GmbH), Marc Harding
=====================================================
Path #1 is: D:" /source3 -vvv, path alias is: /D" /source3 -vvv
Portmap daemon starts failed.
```

And running Net start portmap says no service with that name - strange

**Comment by rfay** [comment]
Reopening now that you're back. It likely can't be something like portmap service, since you're able to get it working with a simple configuration (just C: drive)

**Comment by rfay** [comment]
I tested with a simple approach. I made a D:\junker99 project and used this ~/.ddev/nfs_exports.txt:
```
C:\Users\rfay > /c/Users/rfay
D:\junker99 > /d/junker99
```

After `nssm restart nfsd` it worked perfectly.  

I haven't figured out why you're mounting to /source3 or /seu/source instead of /d/source3

One thing I do remember - a blank line on the end of the file can mess up winnfsd, so you might want to check that.
